pkg/aspect/root/flags/interceptor.go

```diff
@@ -86,32 +86,68 @@ func FlagsInterceptor(streams ioutils.Streams) interceptors.Interceptor {
 		if cfgFile != "" {
 			// Use config file from the flag.
 			viper.SetConfigFile(cfgFile)
+			viper.AutomaticEnv()
+			if err := viper.ReadInConfig(); err == nil {
+				faint.Fprintln(streams.Stderr, "Using config file:", viper.ConfigFileUsed())
+			}
 		} else {
-			// Find home directory.
-			home, err := homedir.Dir()
-			cobra.CheckErr(err)
-
-			// Search for config in home directory with name ".aspect" (without extension).
-			viper.AddConfigPath(home)
+			// used to search in current directory
+			viper.AddConfigPath(".")
 			viper.SetConfigName(".aspect")
 
-			// Search for config in root of current repo with name ".aspect" (without extension).
-			repoViper.AddConfigPath(".")
-			repoViper.SetConfigName(".aspect")
-			repoViper.AutomaticEnv()
-		}
-
-		viper.AutomaticEnv()
-		if err := viper.ReadInConfig(); err == nil {
-			faint.Fprintln(streams.Stderr, "Using config file:", viper.ConfigFileUsed())
-		}
-
-		if err := repoViper.ReadInConfig(); err == nil {
-			faint.Fprintln(streams.Stderr, "Using config file:", repoViper.ConfigFileUsed())
+			// used to search in home directory
+			home, err := homedir.Dir()
+			cobra.CheckErr(err)
+	
+			viper.AutomaticEnv()
+
+			// what do we want to do with error handeling for all of these reads?
+			// it should be consistent just not sure what it should be
+			// the errors that aren't viper.ConfigFileNotFoundError will still be thrown how should they be handled?
+
+			// search in current directory first
+			if err := viper.ReadInConfig(); err != nil {
+				if _, ok := err.(viper.ConfigFileNotFoundError); ok {
+					// search in home directory next
+					viper.AddConfigPath(home)
+					if err := viper.ReadInConfig(); err != nil {
+						if _, ok := err.(viper.ConfigFileNotFoundError); ok {
+							// Config file not found;
+							// create new config file if it does not exist
+							if err := viper.WriteConfigAs(fmt.Sprintf("%s/.aspect.yaml", home)); err != nil {
```

Comment:
It's too aggressive to write files to the user's home directory without asking for permission first.

Reply:
What is the way we would like to handle this? 
My thought was precedence would go to the file in the working directory and then to the file in the home directory. So I was going to write them in the opposite order. Is it preferred to just write to the working directory then?

Reply:
Ide argue that we want to ask the user and then write the to home directory. We need somewhere to have config that is user specific (not repo specific). If we write everything to the repo one then peoples user preferences could get committed when they shouldnt be.

Reply:
Maybe we could add a field to the repo level config along the lines of `userConfig: foo.yaml` and then they could add `foo.yaml` to their `.gitignore` and we could save user specific configs to that file? Then it would never be committed and we dont need to add anything to the home directory? But if they dont have something like that set then we just ask permission to add to the home directory?

Reply:
Also, check the files in your HOME directory. Software add there all the time. Ive got a `.yarn`, `.npm`, `.vscode` in my mine (to name a few) and I didnt give any permission to write to my home folder. Maybe we add a `.aspect` folder where we can add anything we might need such as a config file, logs, etc

Reply:
The most updated version will create a .aspect.yaml file in home directory if others aren't present. It will no longer try to write to the working directory.

If we want to have a user specific file I would stay with similar conventions to bazel and have a user.aspect.yaml and an .aspect.yaml file to make it clear which is which and to make it easy to gitignore for everyone. It would also make it easier for us as we would know the priority of the files beforehand.

The thought behind not asking for permission is to keep some determinism between invocations of the cli. It is also as you pointed out convention now to write config files to the HOME directory, I don't think asking for this permission really adds much and could make it more difficult to use in a CI or automated fashion later on. Also I believe @f0rmiga has changed his opinion from his initial comment but I'll let him clarify.

I don't think adding more complexity of managing different directories is required for this use case. If we want to revisit that for logs or something else that really needs it we definitely can.

Happy to discuss more :)